extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java

```diff
@@ -88,17 +103,29 @@ public OpenCensusProtobufInputRowParser withParseSpec(ParseSpec parseSpec)
       throw new ParseException(e, "Protobuf message could not be parsed");
     }
 
+    // Fetch metric name.
+    String metricName = !Strings.isNullOrEmpty(this.metricDimension) ? this.metricDimension :
+        metric.getMetricDescriptor().getName();
```

Comment:
this should be done at initialization time in the constructor

Reply:
I think we can just have a `boolean` replacing `NullorEmpty` method can be taken in constructor.